### ISSUE TITLE
SMB Server - Exception when client logs in twice.

### ISSUE THREAD

**Issue by Ashthos** [TalAloni/SMBLibrary#146]
Note - I'm aware that clients shouldn't do this, but the server does not handle this scenario gracefully.

When issuing the same `client.Login("domain", "user", "pass")` call twice to the SMB Server, the server crashes with an Argument exception.

`System.ArgumentException: 'An item with the same key has already been added. Key: 1'`

This is thrown from within the following callstack.

```
>	[Exception] System.Private.CoreLib.dll!System.ThrowHelper.ThrowAddingDuplicateWithKeyArgumentException<T>(T key)	Unknown
 	[Exception] System.Private.CoreLib.dll!System.Collections.Generic.Dictionary<TKey, TValue>.TryInsert(TKey key, TValue value, System.Collections.Generic.InsertionBehavior behavior)	Unknown
 	[Exception] SMBLibrary.dll!SMBLibrary.Server.SMB2ConnectionState.CreateSession(ulong sessionID, string userName, string machineName, byte[] sessionKey, object accessToken, bool signingRequired, byte[] signingKey)	Unknown
 	[Exception] SMBLibrary.dll!SMBLibrary.Server.SMB2.SessionSetupHelper.GetSessionSetupResponse(SMBLibrary.SMB2.SessionSetupRequest request, SMBLibrary.Authentication.GSSAPI.GSSProvider securityProvider, SMBLibrary.Server.SMB2ConnectionState state)	Unknown
 	[Exception] SMBLibrary.dll!SMBLibrary.Server.SMBServer.ProcessSMB2Command(SMBLibrary.SMB2.SMB2Command command, SMBLibrary.Server.SMB2ConnectionState state)	Unknown
 	[Exception] SMBLibrary.dll!SMBLibrary.Server.SMBServer.ProcessSMB2Command(SMBLibrary.SMB2.SMB2Command command, ref SMBLibrary.Server.ConnectionState state)	Unknown
 	[Exception] SMBLibrary.dll!SMBLibrary.Server.SMBServer.ProcessSMB2RequestChain(System.Collections.Generic.List<SMBLibrary.SMB2.SMB2Command> requestChain, ref SMBLibrary.Server.ConnectionState state)	Unknown
 	[Exception] SMBLibrary.dll!SMBLibrary.Server.SMBServer.ProcessPacket(SMBLibrary.NetBios.SessionPacket packet, ref SMBLibrary.Server.ConnectionState state)	Unknown
 	[Exception] SMBLibrary.dll!SMBLibrary.Server.SMBServer.ProcessConnectionBuffer(ref SMBLibrary.Server.ConnectionState state)	Unknown
 	[Exception] SMBLibrary.dll!SMBLibrary.Server.SMBServer.ReceiveCallback(System.IAsyncResult result)	Unknown
```

As this is non-user code, we can't catch this exception and the server crashes.

Tested with SMBLibrary 1.4.8 from Nuget.

**Comment by TalAloni** [comment]
I see. I will look into it. thanks.

**Comment by TalAloni** [comment]
This issue is resolved in v1.4.9 - thanks for finding and reporting the issue!
